cmd/servicelog/list.go

```diff
@@ -68,44 +60,18 @@ func init() {
 	listCmd.Flags().BoolVarP(&serviceLogListAllMessagesFlag, "all-messages", "A", serviceLogListAllMessagesFlag, "Toggle if we should see all of the messages or only SRE-P specific ones")
 }
 
-func extractExternalIdFromResponse(response *sdk.Response) (string, error) {
-	status := response.Status()
-	body := response.Bytes()
-
-	if status >= 400 {
-		validateBadResponse(body)
-		log.Fatalf("Failed to list message because of %q", BadReply.Reason)
-		return "", nil
-	}
-
-	validateGoodResponse(body)
-	clusterListGoodReply := servicelog.ClusterListGoodReply{}
-	err := json.Unmarshal(body, &clusterListGoodReply)
-	if err != nil {
-		err = fmt.Errorf("cannot parse good clusterlist response: %w", err)
-		return "", err
-	}
-
-	if clusterListGoodReply.Total != 1 || len(clusterListGoodReply.Items) != 1 {
-		return "", fmt.Errorf("could not find an exact match for the clustername")
+func getClusters(ocmClient *sdk.Connection, clusterIds []string) []*v1.Cluster {
+	for i, id := range clusterIds {
+		clusterIds[i] = fmt.Sprintf(`display_name like '%[1]s' or name like '%[1]s' or id like '%[1]s' or external_id like '%[1]s'`, id)
```

Comment:
Good question, I believe the original reason was that `generateQuery` used the `is` operator: 
```
func generateQuery(clusterIdentifier string) string {
	return strings.TrimSpace(fmt.Sprintf("id like '%[1]s' or external_id like '%[1]s' or display_name like '%[1]s'", clusterIdentifier))
}
```
This prevented wildcard substitution in OCM with the `%` token (ie- only clusters that have an identifier matching `foo%` exactly would be returned by OCM, clusters matching `foobar` would not). 

After taking a second look, I don't see any reason *not* to use `like` in this function - doing so would allow for wildcard expansion with `servicelog post`'s `-p` flag: 
```
~/workspace/osdctl]
[tnierman@tnierman] >> ./dist/osdctl_linux_amd64/osdctl servicelog post -t https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/cluster_has_gone_missing.json -p CLUSTER_UUID="tnierman%" -d
INFO[0000] The following clusters match the given parameters: 
Name                ID                                 State               Version             Cloud Provider      Region
tnierman            <id>                               ready               4.8.3               aws                 us-west-1
tnierman2           <id>                               installing                              aws                 us-west-1

INFO[0000] The following template will be sent:         
{
  "severity": "Error",
  "service_name": "SREManualAction",
  "cluster_uuid": "${CLUSTER_UUID}",
  "summary": "Action required: delete unregistered cluster",
  "description": "Your cluster requires you to take action because it is no longer registering with Red Hat OpenShift Cluster Manager. Possible causes include stopping instances or a networking misconfiguration. If you intended to terminate this cluster then please delete the cluster. Otherwise file a support request.",
  "internal_only": false,
  "event_stream_id": ""
}
```

I've updated list to use `generateQuery`, and modified `generateQuery` to use `like` over `is`, allowing multiple clusters to be messaged using the above syntax